CMakeLists.txt

```diff
@@ -125,13 +129,13 @@ option(
   OFF)
 option(WITH_EXAMPLES "Whether to build examples" ON)
 
-option(WITH_METRICS_PREVIEW "Whether to build metrics preview" OFF)
+option(WITH_METRICS_PREVIEW "Whether to build metrics preview" ON)
```

Comment:
Apologies for enabling metrics and logs preview, just want to make sure that ALL of existing code successfully compiles with the new implementation of variant.

Reply:
We don't need to enable it at CI pipeline: https://github.com/open-telemetry/opentelemetry-cpp/blob/1c70f0e45ea9009c9d5989e3c5cfcf45b8a54004/ci/do_ci.sh#L41
 
Won't that be enough for validation?

Reply:
This is minor thing. I can revert those flags before merge.

Reply:
Fixed.